lib/stm32/common/usart_common_all.c

```diff
@@ -52,7 +52,7 @@ usart_reg_base
 
 void usart_set_baudrate(uint32_t usart, uint32_t baud)
 {
-	uint32_t clock = rcc_apb1_frequency;
+	uint32_t clock = rcc_get_usart_clk_freq(usart);
```

Comment:
you need to delete the lines after this too.... seeing as you are now handling clock selection elsewhere.

I'm guessing you weren't using usart1/6? :)